packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1479,35 +1482,148 @@ export class TransactionController extends BaseController<
         `Cannot update custodial transaction with status: ${status}`,
       );
     }
-    if (status === TransactionStatus.signed) {
-      transactionMeta.status = status;
-    }
+
+    const updatedTransactionMeta = merge(
+      transactionMeta,
+      pickBy({ hash, status }),
+    );
 
     if (status === TransactionStatus.submitted) {
-      transactionMeta.submittedTime = new Date().getTime();
-      transactionMeta.status = status;
+      updatedTransactionMeta.submittedTime = new Date().getTime();
     }
 
     if (status === TransactionStatus.failed) {
-      transactionMeta = {
-        ...transactionMeta,
-        error: normalizeTxError(new Error(errorMessage)),
-        status: TransactionStatus.failed,
-      };
+      updatedTransactionMeta.error = normalizeTxError(new Error(errorMessage));
     }
 
-    if (custodyStatus) {
```

Comment:
I didn't see this change (removing custodyStatus) in the PR description. Is it intentional?

Reply:
Good point, I thought MMI package was updating this property on `handleTxStatusUpdate`

Reply:
This was intentional since I found a few extra properties MMI wanted to set such as `custodyStatusDisplayText` and `custodyId`, plus the validations wouldn't always be applicable since we'd need to set the `custodyId` initially.

This meant it felt like the safer route for now to use this exclusively for updating the hashes and status, but to continue to use the generic `updateTransaction` for their general update to minimise the chance of bugs to begin with.